components/plurals/src/lib.rs

```diff
@@ -550,3 +552,315 @@ impl PluralRules {
             .chain(Some(PluralCategory::Other))
     }
 }
+
+/// A [`PluralRules`] that also has the ability to retrieve an appropriate [`Plural Category`] for a
+/// range.
+///
+/// # Examples
+///
+/// ```
+/// use icu::locid::locale;
+/// use icu::plurals::{PluralCategory, PluralOperands};
+/// use icu::plurals::{PluralRuleType, PluralRulesWithRanges};
+/// use std::convert::TryFrom;
+///
+/// let ranges = PluralRulesWithRanges::try_new(
+///     &locale!("ar").into(),
+///     PluralRuleType::Cardinal,
+/// )
+/// .expect("locale should be present");
+///
+/// let operands =
+///     PluralOperands::try_from(1).expect("Failed to parse to operands.");
+/// let operands2: PluralOperands =
+///     "2.0".parse().expect("Failed to parse to operands.");
```

Comment:
Nit: In new code we use "should" in `expect` messages. Also, `From<usize>` is infallible.

```suggestion
/// let operands = PluralOperands::from(1);
/// let operands2: PluralOperands =
///     "2.0".parse().expect("should parse string to operands");
```

Reply:
I'd actually use `unwrap` here for brevity. It's pretty clear what is happening when reading the code, and the execution is not user-visible. The only reason to use `expect` would be educational, but I'm not sure that's our job.